R/authenticate.R

```diff
@@ -8,15 +10,12 @@
 #'   
 #' @return sets environment variable with login information
 #' @export
-bwg_auth <- function(username, password) {
+bwg_auth <- function() {
   ## should check if it exists, and not rerun the function.
-  token_exists <- exists("token", envir = parent.frame())
-  
-  if (token_exists) stop("you've already got a token")
 
-  ## check inputs
-  assertthat::assert_that(is.character(username))
-  assertthat::assert_that(is.character(password))
+  username <- getPass::getPass("USERNAME: ")
```

Comment:
It uses tcltk only on non-rstudio / non-tty (terminal) inputs. Might be
worth a feature request to getpass to see if drew can make an interface
that will look the same for both requests on all platforms.

On Thu, 10 Mar 2016, 17:20 A.D. Letaw, <notifications@github.com> wrote:

> In R/authenticate.R
> <https://github.com/SrivastavaLab/bwgdbR/pull/2#discussion_r55714915>:
>
> >
> > -  ## check inputs
> > -  assertthat::assert_that(is.character(username))
> > -  assertthat::assert_that(is.character(password))
> > +  username <- getPass::getPass("USERNAME: ")
>
> getPass uses tcltk
> <https://github.com/cran/getPass/blob/master/R/getPass.r#L101> to create
> the pop-up; perhaps you could create your own pop-up to make the username
> input match the password input?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SrivastavaLab/bwgdbR/pull/2/files#r55714915>.
>